tchannel-benchmark/src/main/java/com/uber/tchannel/benchmarks/PingPongServerBenchmark.java

```diff
@@ -74,43 +78,54 @@ public static void main(String[] args) throws RunnerException {
     public void setup() throws Exception {
 
         this.channel = new TChannel.Builder("ping-server")
-                .setMaxQueuedRequests(20000000)
-                .build();
+            .setMaxQueuedRequests(20000000)
+            .build();
         channel.makeSubChannel("ping-server").register("ping", new PingDefaultRequestHandler());
         this.client = new TChannel.Builder("ping-client").build();
         channel.listen();
+        this.subClient = this.client.makeSubChannel("ping-server");
         this.port = this.channel.getListeningPort();
     }
 
     @Benchmark
     @BenchmarkMode(Mode.Throughput)
     public void benchmark(final AdditionalCounters counters) throws Exception {
-
-        Request<Ping> request = new Request.Builder<>(new Ping("ping?"), "some-service", "ping").build();
-
-        ListenableFuture<Response<Pong>> future = this.client.makeSubChannel("ping-server").callJSON(
-            InetAddress.getLocalHost(),
-            this.port,
-            request,
-            Pong.class
-        );
-        Futures.addCallback(future, new FutureCallback<Response<Pong>>() {
+        JsonRequest<Ping> request = new JsonRequest.Builder<Ping>("ping-server", "ping")
+            .setBody(new Ping("ping?"))
+            .build();
+
+        ListenableFuture<JsonResponse<Pong>> future = this.subClient
+            .send(
+                request,
+                InetAddress.getLocalHost(),
+                this.port
+            );
+        Futures.addCallback(future, new FutureCallback<JsonResponse<Pong>>() {
             @Override
-            public void onSuccess(Response<Pong> pongResponse) {
-                counters.actualQPS.incrementAndGet();
+            public void onSuccess(JsonResponse<Pong> pongResponse) {
+
+                if (!pongResponse.isError()) {
+                    counters.actualQPS.incrementAndGet();
+//                    pongResponse.getBody(Pong.class);
```

Comment:
yea, I put it here to show some comparison of lazy evaluation. I will add more comments to explain it.